File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program80.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept N numbers from user and return product of all odd elements.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{
	public int arr[],iCnt,iMulti;

	public Solution(int iNo)
	{
		this.iMulti=1;
		this.iCnt = 0;
		arr = new int[iNo];
	}

	public void Accept()
	{
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter "+arr.length+" elements: ");
		for(iCnt=0;iCnt<arr.length;iCnt++)
		{
			System.out.print("Enter "+ (iCnt+1) + "element: ");
			arr[iCnt] = sObj.nextInt();
		}
	}	

	public int Product()
	{
		for(iCnt=0;iCnt<arr.length;iCnt++)
		{
			if(arr[iCnt]%2!=0)
			{
				iMulti=iMulti*arr[iCnt];
			}
		}

		return iMulti;
	}

}

class Program
{
	public static void main(String[] args)
	{
		int iRet = 0;
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter number of elements:");
		int iLength = sObj.nextInt();

		Solution obj = new Solution(iLength);
		obj.Accept();
		iRet = obj.Product();
	
		System.out.println(iRet);
	}
}

/*
OUTPUT:
Enter number of elements:
6
Enter 6 elements: 
Enter 1element: 15
Enter 2element: 66
Enter 3element: 3
Enter 4element: 70
Enter 5element: 10
Enter 6element: 88
45


*/

File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program80.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept N numbers from user and return product of all odd elements.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////



#include<iostream>
using namespace std;

class Solution
{
	private:
		int *iArr;
		int iSize,iCnt,iNum;

	public:
		Solution(int iNo=0,int iNum=0,int iCnt=0)
		{
			this->iNum=iNum;
			this->iSize=iNo;
			iArr = new int[iSize];
		}

		void Accept()
		{
			cout<<"Enter "<<iSize<<" elements:\n";
			for(iCnt=0;iCnt<iSize;iCnt++)
			{
				cout<<"Enter "<<iCnt+1<<" element: ";
				cin>>iArr[iCnt];
			}
		}
		
		int Product()
		{
			int iMulti=1;
			for(iCnt=0;iCnt<iSize;iCnt++)
			{
				if(iArr[iCnt]%2!=0)
				{
					iMulti=iMulti*iArr[iCnt];
				}
				
			}

			return iMulti;
		}

		~Solution()
		{
			delete []iArr;
		}

};

int main()
{
	int iLength=0,iRet=0;

	cout<<"Enter number of elements: ";
	cin>>iLength;

	Solution *dObj = new Solution(iLength);

	dObj->Accept();
	iRet = dObj->Product();

	if(iRet==1)
	{
		cout<<"0\n";
	}
	else
	{
		cout<<iRet<<"\n";
	}

	delete dObj;
	return 0;
}

/*
OUTPUT:
Enter number of elements: 6
Enter 6 elements:
Enter 1 element: 15
Enter 2 element: 66
Enter 3 element: 3
Enter 4 element: 70
Enter 5 element: 10
Enter 6 element: 88
45
Enter number of elements: 6
Enter 6 elements:
Enter 1 element: 44
Enter 2 element: 66
Enter 3 element: 72
Enter 4 element: 70
Enter 5 element: 10
Enter 6 element: 88
0

*/
